src/unix/apple/macosx/hid.c

```diff
@@ -164,15 +165,28 @@ struct hid *mty_hid_create(HID_CONNECT connect, HID_DISCONNECT disconnect, HID_R
 	CFArrayRef matches = CFArrayCreate(kCFAllocatorDefault, (const void **) dict_list, key ? 4 : 3, NULL);
 	IOHIDManagerSetDeviceMatchingMultiple(ctx->mgr, matches);
 
+	IOHIDManagerScheduleWithRunLoop(ctx->mgr, CFRunLoopGetCurrent(), kCFRunLoopDefaultMode);
+
+	IOReturn e = IOHIDManagerOpen(ctx->mgr, kIOHIDOptionsTypeNone);
+	if (e == kIOReturnExclusiveAccess && key) {
+		CFMutableDictionaryRef d4 = IOServiceNameMatching("org_pqrs_Karabiner_DriverKit_VirtualHIDKeyboard");
+		CFMutableDictionaryRef dict_list2[] = {d0, d1, d2, d4};
+
+		CFArrayRef matches2 = CFArrayCreate(kCFAllocatorDefault, (const void **) dict_list2, 4, NULL);
+		IOHIDManagerSetDeviceMatchingMultiple(ctx->mgr, matches);
```

Comment:
why did you make `matches2`? you don't use it for anything.

Reply:
Resolved this, and added a log line for when compatibility is used successfully